modules/ROOT/pages/errors/index.adoc

```diff
@@ -1,92 +1,200 @@
+Status codes can also indicate transient problems that may go away if you retry the request.
+The server error classification determines the effect on the transaction.
+
+.Server error classification
+[options="header", cols="<1m,<2,<1"]
+|===
+
+| Classification
+| Description
+| Effect on the transaction
+
+| xref:errors/all-errors.adoc#_client_errors[ClientError]
+| These errors are caused by the client and are usually related to the request itself.
+Changing the request might yield a successful outcome.
+Legacy client errors have the prefix `Neo.ClientError`.
+| Rollback
+
+| xref:errors/all-errors#_transient_errors[TransientError]
+| These errors are detected by the server and are usually related to some kind of database unavailability, such as limits reached, out-of-memory, timeouts, etc.
+The error can be temporary, therefore retrying later might yield a successful outcome.
+Legacy transient errors have the prefix Neo.`TransientError`.
+| Rollback
+
+| xref:errors/all-errors#_database_error[DatabaseError]
+| These errors are caused by the database and are usually related to the database state and mean that the database failed to service the request.
+Legacy database errors have the prefix `Neo.DatabaseError`.
+| Rollback
+
+|===
+
+[[error-internals]]
+== Error internals
+
+Neo4j supports server errors in the form of Java exceptions.
+Most of these implement the `HasStatus` interface, which means they have a Neo4j status code in addition to the exception message.
+
+On the server side, an exception contains normal Java constructors and methods like `getMessage()`,  `getCause()`, etc., and additionally the `status()` method from the `HasStatus` API, which returns the Neo4j status code.
+
+Starting from 5.25, the exceptions also get new compulsary fields for `gqlStatus` and `statusDescription`, and optional fields for the `diagnosticRecord` and `cause`.
+The cause field is a wrapped underlying Neo4j exception, which in turn has its own GQLSTATUS, statusdescription, diagnostic record and message. +
+The `getMessage()` method is kept as Java exceptions inherently have this method.
+And a new classification field is added to cover the division of client errors, transient errors and database errors, which today is part of the Neo4j status code.
+All of these fields construct the GQLSTATUS object, which is sent to the driver as part of the Failure Bolt message.
+Exactly how this looks, depends on the combination of driver and server versions.
+See <<server-driver-compatibility, Server-driver version compatibility>> for more information.
+
+On the driver side, the Neo4jException is extended with the corresponding methods as on the server side.
+The driver receives the failure message and extracts the status code and the error message.
+Then, it constructs an error object with the status code, error message, and other relevant information, and sends it to the client.
```

Comment:
Ask @bigmontz whether this sentence is correct.

Reply:
I think it is pretty much it. I don't know if it worth to mention that Neo4jException can also be generated on the driver side when drivers can't establish connection to the database, those errors was previously mapped as `ServiceUnavailable` and `SessionExpired` as neo4j code and they received status `50N42` for now since we don't have status code for them (and I don't know if we will have it).